lib/bozo/preparers/sql_server.rb

```diff
@@ -0,0 +1,94 @@
+module Bozo::Preparers
+
+  # Executes a sql script
+  #
+  # == Hook configuration
+  #
+  #   prepare :sql_server do |t|
+  #     t.variable 'DatabaseName', 'MyDatabase'
+  #     t.variable 'MySecondVariable', 'HelloWorld'
+  #     t.script 'my/specific/script.sql' # must be a specific file
+  #     t.connection_string { |c| c[:config_value] } # a function taking the configuration as an argument
+  #   end
+  #
+  # The variables are accessible via `$(KEY)` in the sql script.
+  class SqlServer
+
+    def execute
+      configuration = load_config
+
+      execute_command :sqlcmd, generate_args(configuration)
+    end
+
+    def variable(key, value)
+      @variables ||= {}
+      @variables[key] = value
+    end
+
+    def script(value)
+      @script = value
+    end
+
+    def database_name(name)
```

Comment:
Not keen on adding `database_name` which is only used with the default script.

How about the following, and sniff the `@script` for the symbol.

```ruby
prepare :sql_server do |f|
   f.variable 'DATABASE_NAME', 'MyDb'
   f.script :create_db
   f.connection_string { |c| c[:connection] }
end
```